packages/flutter/lib/src/material/expansion_panels.dart

```diff
@@ -0,0 +1,294 @@
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'expand_icon.dart';
+import 'mergeable_material.dart';
+import 'theme.dart';
+
+const double _kPanelHeaderCollapsedHeight = 48.0;
+const double _kPanelHeaderExpandedHeight = 64.0;
+
+/// This callback gets called when [ExpansionPanel] with index [panelIndex] gets
+/// expanded or collapsed. In the case where it is expanded, [isExpanded] will
+/// be true.
+typedef void ExpansionPanelCallback(int panelIndex, bool isExpanded);
+
+/// This widget builder is useful when the header of the [ExpansionPanel] needs
+/// react to expansion.
+///
+/// An example would be to use an [AnimatedCrossfade] that triggers when
+/// [isExpanded] changes.
+typedef Widget ExpansionPanelHeaderBuilder(BuildContext context, bool isExpanded);
+
+/// A Material expansion panel. It has a header and a body and can be either
+/// expanded or collapsed. The body of the panel is only visible when it is
+/// expanded.
+///
+/// Expansion panels are used as children for [ExpansionPanelList].
+///
+/// See also:
+///
+///  * [ExpansionPanelList]
+///  * <https://material.google.com/components/expansion-panels.html>
+class ExpansionPanel {
+  /// Creates an expansion panel to be used as a child for [ExpansionPanelList].
+  ExpansionPanel({
+    this.headerBuilder,
+    this.body,
+    this.isExpanded: false
+  }) {
+    assert(this.headerBuilder != null);
+    assert(this.body != null);
+    assert(this.isExpanded != null);
+  }
+
+  /// The widget builder that builds the header part of the expansion panel
+  /// while providing the expansion state of the current panel.
+  ///
+  /// This widget is always visible.
+  final ExpansionPanelHeaderBuilder headerBuilder;
+
+  /// The body of the expansion panel that site below the header.
+  ///
+  /// This widget is visible only when the panel is expanded.
+  final Widget body;
+
+  /// Whether the panel is expanded.
+  ///
+  /// Defaults to false.
+  final bool isExpanded;
+}
+
+/// A Material expansion panel list that lays out its children and animates
+/// expansions.
+///
+/// See also:
+///
+///  * [ExpansionPanel]
+///  * <https://material.google.com/components/expansion-panels.html>
+class ExpansionPanelList extends StatefulWidget {
+  /// Creates an expansion panel list widget. The [expansionCallback] is
+  /// triggered whenever one of the expansion icon is pushed.
+  ExpansionPanelList({
+    Key key,
+    this.children: const <ExpansionPanel>[],
+    this.expansionCallback
+  }) : super(key: key) {
+    assert(this.children != null);
+  }
+
+  /// The children of the expansion panel list. They are layed in a similar
+  /// fashion to [BlockBody].
+  final List<ExpansionPanel> children;
+
+  /// The callback that gets called whenever one of the expansion icons is
+  /// pressed. The arguments passed to the callback are the index of the
+  /// to-be-expanded panel in the list and whether the panel is currently
```

Comment:
the index of the to-be-expanded panel in the list => the to-be-expanded panel's index